readthedocs/proxito/views/serve.py

```diff
@@ -118,6 +125,17 @@ def get(self,
         )
 
         is_external = unresolved_domain.is_from_external_domain
+        if (
+            is_external
+            and original_version_slug
+            and original_version_slug != version_slug
+        ):
+            raise Http404("Version doesn't match the version from the domain.")
+
+        manager = EXTERNAL if is_external else INTERNAL
+        version = (
+            final_project.versions(manager=manager).filter(slug=version_slug).first()
+        )
```

Comment:
Is this change just pulling in the recent security fix?

Reply:
yeah, looks like you were looking at the merge commit, this isn't in the final diff :D